### PR TITLE
Update prow jobs to work with both pod-utils and current image

### PR DESCRIPTION
**Pull request by sebastienvas**
Update prow jobs to work with both pod-utils and current image to ease update.

Old Prow istio image used embedded boostrap code. The pod-utils feature, externalize the checkout and log gathering, meaning the test container can focus on tools related to build and test istio.

More info about [pod-utils](https://github.com/kubernetes/test-infra/blob/master/prow/pod-utilities.md)

### CODE REVIEW

File: prow/lib.sh [hunk 1]
```diff
@@ -15,17 +15,26 @@
...
+      fi
+
+      # Set artifact dir based on checkout
+      export ARTIFACTS_DIR="${GOPATH}/src/istio.io/istio/_artifacts"
+    fi
+
+    if [[ "${CI:-}" == 'prow' ]]; then
+      # Set artifact dir based on checkout
+      export ARTIFACTS_DIR="${ARTIFACTS}"
     fi
```
Comment (hklai) [inline prow/lib.sh]: Perhaps a final catch all case in case $CI is messed up?

File: prow/lib.sh [hunk 2]
```diff
@@ -15,17 +15,26 @@
...
+        mv "${GOPATH}/src/github.com/${REPO_OWNER:-istio}" "${GOPATH}/src/istio.io"
+        export ROOT=${GOPATH}/src/istio.io/istio
+        cd "${GOPATH}/src/istio.io/istio" || return
+      fi
+
+      # Set artifact dir based on checkout
+      export ARTIFACTS_DIR="${GOPATH}/src/istio.io/istio/_artifacts"
+    fi
+
+    if [[ "${CI:-}" == 'prow' ]]; then
```
Comment (hklai) [inline prow/lib.sh]: elseif?

File: prow/lib.sh (hunk 1 above)
Comment (sebastienvas) [reply prow/lib.sh]: We are just setting up ARTIFACTS_DIR here. If CI is messed up then this won't be set.

File: prow/lib.sh (hunk 2 above)
Comment (sebastienvas) [reply prow/lib.sh]: good point

**Comment by sebastienvas** [comment]
I ll be testing this with https://github.com/istio/istio/pull/8476

**Pull request merged**